File: leetcode_2021/leetcode_go/1.两数之和.go
func twoSum(nums []int, target int) []int {
    d := make(map[int]int)
    for i := 0; i < len(nums); i ++ {
        another := target - nums[i]
        if idx, ok := d[another]; ok {
            return []int{idx, i}
        }
        d[nums[i]] = i
    }
    return nil
}

File: leetcode_2021/leetcode_cpp/1.两数之和.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        unordered_map<int, int> d;
        for (int i = 0; i < sizeof(nums); i++) {
            auto iter = d.find(target - nums[i]);
            if (iter != d.end()) {
                return {d[target - nums[i]], i};
            }
            d[nums[i]] = i;
        }
        return {};
    }
};